src/pages/home/report/ReportActionCompose.js

```diff
@@ -78,12 +73,9 @@ class ReportActionCompose extends React.Component {
             this.comment = this.props.comment;
         }
 
-        // When any modal goes from visible to hidden or when the report ID changes, bring focus to the compose field
-        if (
-            (prevProps.modal.isVisible && !this.props.modal.isVisible)
-            || (prevProps.reportID !== this.props.reportID)
-        ) {
-            this.setIsFocused(true);
+        // When the report ID changes bring focus to the compose field
+        if (prevProps.reportID !== this.props.reportID) {
```

Comment:
Hey reviewers I think we might have a merge conflict here as @parasharrajat  work needs to change this line as well #2022 

I've changed it, to exclude modal checks as they break modal to modal transition - the attachment modal would close immediately if focus is requested on the field 

If I revert this change, I'll also need to bring  the timeout back 😭

Any suggestions?

Reply:
Hmm, I've actually can revert most of my changes. The UX would be a bit odd but I guess this will be improved once autofocussing is solved and the keyboard stays collapsed

Here are some samples:
#### reverting almost everything in `ReportActionCompose` to original 

##### On Simulator
https://user-images.githubusercontent.com/12156624/112962392-022ed680-914f-11eb-9158-f53e76a1a36a.mp4

##### On a Real Device

https://user-images.githubusercontent.com/12156624/112963857-7453eb00-9150-11eb-99e3-ecc842168137.mp4

There's also a video showing how the current changes work in the pull request description

Anyway I'm willing to revert my changes to  `ReportActionCompose` so that we can merge this easier if another ticket is already addressing focusing

Reply:
The other PR should actually be changing to solve the focus issue in a less general way, so the modal checks you mention won't be a conflict for much longer.

I think this is fine for now actually.

Reply:
Hmm, maybe this is required after all for the regression I [mentioned here](https://github.com/Expensify/Expensify.cash/pull/2114#issuecomment-810095724)

Reply:
Changes reverted.